pipeline/container.go

```diff
@@ -82,48 +83,87 @@ func (c *ContainerSlice) Purge(r *RuleData) *ContainerSlice {
 func (c *ContainerSlice) Sanitize(driver string) *ContainerSlice {
 	containers := new(ContainerSlice)
 
+	// iterate through each Container in the pipeline
+	for _, container := range *c {
+		// sanitize container
+		cont := container.Sanitize(driver)
+
+		// append the Container to the new slice of Containers
+		*containers = append(*containers, cont)
+	}
+
+	return containers
+}
+
+// Sanitize cleans the fields for every step in the pipeline so they
+// can be safely executed on the worker. The fields are sanitized
+// based off of the provided runtime driver which is setup on every
+// worker. Currently, this function supports the following runtimes:
+//
+//   * Docker
+//   * Kubernetes
+func (c *Container) Sanitize(driver string) *Container {
```

Comment:
Good idea! I like this better 👍

Reply:
In the future, I think we could do this to the `Purge` function as well to break the code up into smaller functions.